zk/legacy_executor_verifier/legacy_executor_verifier.go

```diff
@@ -197,45 +234,59 @@ func (v *LegacyExecutorVerifier) AddRequestUnsafe(ctx context.Context, tx kv.RwT
 		// 	atomic.StoreInt32(&counter, 1)
 		// }
 
-		response := &VerifierResponse{
+		return NewVerifierBundle(request, &VerifierResponse{
 			BatchNumber:      request.BatchNumber,
 			Valid:            ok,
 			Witness:          witness,
 			ExecutorResponse: executorResponse,
 			Error:            executorErr,
-		}
-
-		return response, nil
+		}), nil
 	})
 
 	// add batch to the list of batches we've added
 	v.addedBatches[request.BatchNumber] = struct{}{}
 
 	// add the promise to the list of promises
 	v.promises = append(v.promises, promise)
-	return promise, nil
+	return promise
 }
 
-func (v *LegacyExecutorVerifier) ConsumeResultsUnsafe(tx kv.RwTx) ([]*VerifierResponse, error) {
+// Unsafe is not thread-safe so it MUST be invoked only from a single thread
+func (v *LegacyExecutorVerifier) ProcessResultsUnsafe(tx kv.RwTx) ([]*VerifierResponse, error) {
 	hdb := hermez_db.NewHermezDbReader(tx)
 
 	results := make([]*VerifierResponse, 0, len(v.promises))
-	for _, promise := range v.promises {
-		result, err := promise.GetNonBlocking()
-		if result == nil && err == nil {
+	for i := 0; i < len(v.promises); i++ {
+		verifierBundle, err := v.promises[i].TryGet()
+		if verifierBundle == nil && err == nil {
 			break
 		}
+
+		verifierResponse := verifierBundle.response
 		if err != nil {
-			log.Error("error getting verifier result", "err", err)
+			log.Error("error on our end while preparing the verification request, re-queueing the task", "err", err)
+			// this is an error on our end, so just re-create the promise at exact position where it was
+			if verifierBundle.request.isOverdue() {
+				return nil, fmt.Errorf("error: batch %d couldn't be processed in 30 minutes", verifierBundle.request.BatchNumber)
```

Comment:
do we want to return here or just log and requeue it again?

Reply:
yes, we must return here, because if batch N is not process then we cannot process the results of any batch > N so practically no point to wait/check for any batch number > n

Reply:
Hmmm so in 30 minutes the process can't confirm one batch, which will cause all subsequent batches to not get verified also.... is this a panic worthy state to be in?

Reply:
Tbh - I have no idea what to do if a batch cannot be confirmed in 30min. Any ideas?

Reply:
Let's not let this decision hamper getting this merged